python/cudf_polars/pyproject.toml

```diff
@@ -115,6 +114,9 @@ ignore = [
   "TD003", # Missing issue link on the line following this TODO
   # tryceratops
   "TRY003", # Avoid specifying long messages outside the exception class
+  # pyupgrade
+  "UP035",  # Import from `collections.abc` instead: `Callable`
```

Comment:
Everything else in `cudf` uses `typing.Callable`.

```shell
git grep -E 'import.*Callable'
```

Reply:
Confused by this one, is this an auto-upgrade fix that ruff can do for us?

Reply:
I'm not sure if this is one `ruff` will do automatically if asked.

But either way... it recommended this for `cudf_polars` but not for the rest of `cudf`, because `cudf_polars` has its own `ruff` config separate from the rest of `cudf`. 

`cudf_polars` uses all the `pyupgrade` checks

https://github.com/rapidsai/cudf/blob/8d6b2616af8aeec6dfd02d787084c583e2447791/python/cudf_polars/pyproject.toml#L89

All of the other Python code in the repo only has specific `pyupgrade` checks run over it

https://github.com/rapidsai/cudf/blob/8d6b2616af8aeec6dfd02d787084c583e2447791/pyproject.toml#L87-L90

Happy to do whatever you want me to with this particular check, I don't have a strong preference for either pattern so I chose the one that would at least be consistent with the rest of the Python code in this repo.

Reply:
[`typing.Callable`](https://docs.python.org/3/library/typing.html#typing.Callable) is the deprecated alias, so in cudf-polars can we migrate to using `collections.abc.Callable`?
I realise in cudf-classic this is more work, so we can defer doing it there.

Reply:
I would propose deferring this change to a separate PR unless we can knock it out today. Completing the drop of Python 3.9 is important so we can start adding Python 3.12.

Reply:
Ok sure, happy to do that in a follow-up PR after this.

Reply:
> *in a follow-up PR after this*

Put up #16670 making this change across `cudf`.